core/trino-main/src/main/java/io/trino/execution/buffer/CompressionCodec.java

```diff
@@ -0,0 +1,19 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.execution.buffer;
+
+public enum CompressionCodec
+{
+    LZO, SNAPPY, LZ4, ZSTD
```

Comment:
What's the rationale for supporting LZO and SNAPPY here ?
LZ4 offers less aggressive compression at small CPU cost while ZSTD offers aggressive compression at high CPU cost. LZO and SNAPPY are somewhere in between, when would we use either of those two ? We support them in file formats because the standard for those formats requires it. For our own internal usage we should offer fewer choices which fit some use case.

Reply:
Agreed. LZ4 should perform better and compress as well as SNAPPY. In other words, LZ4 should be strictly superior to SNAPPY, so there is no need to support it.